[docs] Refactor DVP getting started guide

prismagod opened the pull request:
## Description

Redesigned and extensively refactored the DVP Getting Started guide into a shorter, interactive flow. The installation is reduced from 10 steps to 6, and parameters entered in the form are automatically substituted into commands and configuration snippets.

Main changes:

- Added validated fields for network, node, DNS, and NFS parameters.
- Added automatic generation of `config.yml`, administrator credentials, and the worker enrollment SSH key.
- Combined storage, virtualization, ingress, console, and access setup into the bootstrap configuration.
- Added cluster verification, DNS setup, and web interface access instructions.
- Updated English and Russian documentation and added an architecture diagram.
- Removed a substantial amount of legacy logic, obsolete steps, and duplicated configuration fragments.

## Why do we need it, and what problem does it solve?

The previous ten-step guide relied on legacy, fragmented logic and required users to copy values manually between commands. The refactored flow collects and validates parameters once, generates a consistent configuration, and removes obsolete code and content, making the quick start shorter and easier to maintain.

## Why do we need it in the patch release (if we do)?

<!---
Describe why the changes need to be backported into the patch release.

If it doesn't matter whether the changes will be backported into the patch release, specify "Not necessarily".

Delete the section if the PR is for release, and not for the patch release.
-->

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries
<!---
  Describe the changes so they will be included in a release changelog.

  Find examples and documentation below, or visit the [Guidelines for working with PRs](https://github.com/deckhouse/deckhouse/wiki/Guidelines-for-working-with-PRs).
-->

```changes
section: docs
type: feature
summary: Redesigned and refactored the DVP Getting Started guide.
impact_level: low
```

<!---
`impact_level: default` adds to changelog as usual, this is the default that can be omitted
`impact_level: high`    something important for users, the impact will be copied to "Know Before Update" section
`impact_level: low`     omitted in changelog YAML; note there is `type:chore` for chores

Tip for the section field:

  - <kebab-case of a module>, e.g. "cloud-provider-aws", "node-manager"
  - "ci", has forced low impact
  - "docs", includes website changes, should have low impact
  - "candi"
  - "deckhouse-controller"
  - "dhctl"
  - "global-hooks"
  - "go_lib"
  - "helm_lib"
  - "jq_lib"
  - "shell_lib"
  - "testing", has forced low impact
  - "tools", has forced low impact

Find changed sections:

gh pr diff   $PULL_REQUEST_NUMBER   |
  egrep "^([+]{3} b|[-]{3} a)/" |
  cut -d/ -f2- |
  sed 's#^ee/##' |
  sed 's#^fe/##' |
  sed 's#^modules/##' |
  sed 's#[0-9][0-9][0-9]-##' |
  egrep -v 'Makefile' |       # add file exclusion here
  cut -d/ -f1 |
  sort |
  uniq

Find all possible sections (excluding ci):

node -e 'console.log(require("./.github/scripts/js/changelog-find-sections.js")().join("\n"))'
-->

prismagod commented on docs/site/_includes/getting_started/dvp/bm/STEP_INSTALL_SCHEMA_RU.md:
```diff
@@ -1,45 +1,107 @@
 
-## Что необходимо для установки
+## Требования к аппаратному и программному обеспечению
+
```
```suggestion

Для установки Deckhouse Virtualization Platform необходима корректная настройка следующих компонентов:

```

mortelumina98 commented on docs/site/_includes/getting_started/dvp/bm/STEP_INSTALL_SCHEMA_RU.md:
```diff
@@ -1,45 +1,108 @@
+
+<ol>
+
+  <li><p><strong>Персональный компьютер</strong> — машина для установки, с которой будет производиться установка Deckhouse Virtualization Platform. Персональный компьютер нужен только для запуска инсталлятора и не будет частью кластера.</p>
```
```suggestion
  <li><p><strong>Персональный компьютер</strong> — машина для установки, с которой будет производиться установка Deckhouse Virtualization Platform. Она используется только для запуска инсталлятора и не входит в состав кластера.</p>
```

mortelumina98 commented on docs/site/_includes/getting_started/dvp/bm/STEP_INSTALL_SCHEMA_RU.md:
```diff
@@ -1,45 +1,108 @@
+  - systemd версии `244`;
+  - поддержка модуля ядра `erofs`.
+  
+  Некоторые дистрибутивы (например, Astra Linux 1.7.4) не соответствуют этим требованиям, и ОС на узлах необходимо привести в соответствие требованиям перед установкой Deckhouse Virtualization Platform. Подробнее — [в документации](/products/kubernetes-platform/documentation/v1/reference/api/cr.html#clusterconfiguration-defaultcri).
```
```suggestion
  Некоторые дистрибутивы (например, Astra Linux 1.7.4) не соответствуют этим требованиям, и ОС на узлах необходимо привести в соответствие с требованиями перед установкой Deckhouse Virtualization Platform. Подробнее — [в документации](/products/kubernetes-platform/documentation/v1/reference/api/cr.html#clusterconfiguration-defaultcri).
```

mortelumina98 commented on docs/site/_includes/getting_started/dvp/bm/STEP_INSTALL_SCHEMA_RU.md:
```diff
@@ -1,45 +1,108 @@
+    - ядро Linux версии `5.8` или новее.
+  - **Программное обеспечение**:
+    - установленные пакеты `cloud-init` и `cloud-utils` (названия могут различаться в зависимости от дистрибутива);
+    - на узле не должно быть установлено пакетов container runtime, например containerd или Docker.
```
```suggestion
    - на узле не должно быть установлено пакетов container runtime, например `containerd` или `Docker`.
```

mortelumina98 replied:
пакеты вроде в бэктиках

mortelumina98 commented on docs/site/_includes/getting_started/dvp/bm/STEP_INSTALL_SCHEMA_RU.md:
```diff
@@ -1,45 +1,108 @@
+    - **уникальный hostname** в пределах серверов (виртуальных машин) кластера.
+
+  {% alert level="warning" %}
+  Обеспечение стабильной работы механизмов живой миграции требует использования идентичной версии ядра Linux на всех узлах кластера.
```
```suggestion
  Для стабильной работы живой миграции требуется использование идентичной версии ядра Linux на всех узлах кластера.
```

mortelumina98 commented on docs/site/_includes/getting_started/dvp/bm/STEP_INSTALL_SCHEMA_RU.md:
```diff
@@ -1,45 +1,108 @@
+  - **Системные требования**:
+    - [поддерживаемая ОС](/products/kubernetes-platform/documentation/v1/supported_versions.html) с возможностью установки NFS-сервера;
+    - достаточный объём дискового пространства под диски виртуальных машин.
+  - **Требования по доступам и сети**:
```
```suggestion
  - **Доступы и сеть**:
```

mortelumina98 commented on docs/site/_includes/getting_started/dvp/bm/STEP_INSTALL_SCHEMA_RU.md:
```diff
@@ -1,45 +1,108 @@
+  - **Требования по доступам и сети**:
+    - сетевой доступ по протоколу NFS (рекомендуется NFSv4.1) с master- и worker-узлов;
+    - экспорт каталога для DVP с опцией `no_root_squash`;
+    - SSH-доступ от **персонального компьютера** (см. п.1) по ключу для администрирования NFS-сервера.
```
```suggestion
    - SSH-доступ от **персонального компьютера** (см. п. 1) по ключу для администрирования NFS-сервера.
```

mortelumina98 commented on docs/site/_includes/getting_started/dvp/bm/STEP_INSTALL_SCHEMA_RU.md:
```diff
@@ -1,45 +1,108 @@
 
-   Требования аналогичны требованиям к master-узлу, но также зависят от характера запускаемой на узлах нагрузки.
-   Также на worker-узлах требуются дополнительные диски для развертывания программно определяемого хранилища, как минимум один диск без разметки емкостью не менее 200 ГБ.
+Для операционных систем семейства Windows платформа поддерживает инициализацию с помощью [autounattend](https://learn.microsoft.com/ru-ru/windows-hardware/manufacture/desktop/windows-setup-automation-overview) установки.
```
```suggestion
Для операционных систем семейства Windows платформа поддерживает инициализацию с помощью [autounattend установки](https://learn.microsoft.com/ru-ru/windows-hardware/manufacture/desktop/windows-setup-automation-overview).
```

mortelumina98 commented on docs/site/_includes/getting_started/dvp/global/STEP_FINALIZE_RU.md:
````diff
@@ -0,0 +1,253 @@
+   sudo -i d8 k get module csi-nfs -w
+   ```
+
+1. Создайте ресурс NFSStorageClass, который описывает подключение к вашему NFS-серверу:
````
```suggestion
1. Создайте ресурс [NFSStorageClass](/modules/csi-nfs/cr.html#nfsstorageclass), который описывает подключение к вашему NFS-серверу:
```

deckhouse-BOaTswain commented:
:red_circle: `Deploy web to stage` for `deckhouse:docs/dvp-gs-update` [failed](https://github.com/deckhouse/deckhouse/actions/runs/20332140311) in 2m8s.

<details><summary>Environment URLS</summary>
<ul><li>Stage: <a href="https://deckhouse.stage.flant.com/products/kubernetes-platform/documentation/v1/">deckhouse.stage.flant.com</a></li><li>Test: <a href="https://deckhouse.test.flant.com/products/kubernetes-platform/documentation/v1/">deckhouse.test.flant.com</a></li><li>Test2: <a href="https://deckhouse.2.test.flant.com/products/kubernetes-platform/documentation/v1/">deckhouse.2.test.flant.com</a></li><li>Test3: <a href="https://deckhouse.3.test.flant.com/products/kubernetes-platform/documentation/v1/">deckhouse.3.test.flant.com</a></li><li>Test4: <a href="https://deckhouse.4.test.flant.com/products/kubernetes-platform/documentation/v1/">deckhouse.4.test.flant.com</a></li><li>Test5: <a href="https://deckhouse.5.test.flant.com/products/kubernetes-platform/documentation/v1/">deckhouse.5.test.flant.com</a></li><li>Test6: <a href="https://deckhouse.6.test.flant.com/products/kubernetes-platform/documentation/v1/">deckhouse.6.test.flant.com</a></li><li>Test7: <a href="https://deckhouse.7.test.flant.com/products/kubernetes-platform/documentation/v1/">deckhouse.7.test.flant.com</a></li></ul></details>

deckhouse-BOaTswain commented:
Aye, aye, @prismagod. I've started the workflow for label 'deploy/web/stage'...
<!-- workflow_start -->
  :fast_forward: `Deploy web to stage` for `deckhouse:docs/dvp-gs-update` [started](https://github.com/deckhouse/deckhouse/actions/runs/20332370998).

deckhouse-BOaTswain commented:
:green_circle: `Deploy web to test10` for `deckhouse:docs/dvp-gs-update` [succeeded](https://github.com/deckhouse/deckhouse/actions/runs/26163604209) in 10m28s.

Environment URL: <a href="https://deckhouse-10.test.flant.dev/products/kubernetes-platform/documentation/v1/">deckhouse-10.test.flant.dev</a>

prismagod committed:
docs: update DVP getting started guide

Signed-off-by: Vladislav Panfilov <vladislav.panfilov@flant.com>

deckhouse-BOaTswain commented:
:green_circle: `Deploy web to test10` for `deckhouse:docs/dvp-gs-update` [succeeded](https://github.com/deckhouse/deckhouse/actions/runs/27021892731) in 6m31s.

Environment URL: <a href="https://deckhouse-10.test.flant.dev/products/kubernetes-platform/documentation/v1/">deckhouse-10.test.flant.dev</a>

deckhouse-BOaTswain commented:
:green_circle: `Deploy web to test10` for `deckhouse:docs/dvp-gs-update` [succeeded](https://github.com/deckhouse/deckhouse/actions/runs/27122531471) in 2m10s.

Environment URL: <a href="https://deckhouse-10.test.flant.dev/products/kubernetes-platform/documentation/v1/">deckhouse-10.test.flant.dev</a>

prismagod committed:
Fix GS 1 next button

Signed-off-by: Vladislav Panfilov <vladislav.panfilov@flant.com>

deckhouse-BOaTswain commented:
:green_circle: `Deploy web to test10` for `deckhouse:docs/dvp-gs-update` [succeeded](https://github.com/deckhouse/deckhouse/actions/runs/27124986246) in 2m30s.

Environment URL: <a href="https://deckhouse-10.test.flant.dev/products/kubernetes-platform/documentation/v1/">deckhouse-10.test.flant.dev</a>

deckhouse-BOaTswain commented:
:green_circle: `Deploy web to test10` for `deckhouse:docs/dvp-gs-update` [succeeded](https://github.com/deckhouse/deckhouse/actions/runs/27953046306) in 4m25s.

Environment URL: <a href="https://deckhouse-10.test.flant.dev/products/kubernetes-platform/documentation/v1/">deckhouse-10.test.flant.dev</a>

Zhbert committed:
Updates

Signed-off-by: Zhbert <konstantin.nezhbert@flant.com>

Zhbert reviewed (approved):
Everything works, I checked it on a real cluster.

One small improvement suggestion: if the user has an older version of Firefox or Chrome, the SSH key ed25519 will not be generated in Web Crypto, and further steps will not work (there will be empty fields). Here I would like to add a check for the key generation and show an error if it did not work out. 

Something like:
```js
(async () => {
  try {
    const kp = await crypto.subtle.generateKey({ name: 'Ed25519' }, true, ['sign', 'verify']);
    console.log('Ed25519 OK', kp);
  } catch (e) {
    console.error('Ed25519 NOT supported', e);
  }
})();
```
and show an alert with the minimum required browser versions:
Chrome / Edge 113+
Firefox 129+
Safari 17+ (macOS / iOS) 

[Table with versions](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/generateKey#browser_compatibility)

Zhbert committed:
Updates

Signed-off-by: Zhbert <konstantin.nezhbert@flant.com>

Zhbert committed:
Updates

Signed-off-by: Zhbert <konstantin.nezhbert@flant.com>

deckhouse-BOaTswain commented:
:green_circle: `Deploy web to test4` for `deckhouse:docs/dvp-gs-update` [succeeded](https://github.com/deckhouse/deckhouse/actions/runs/29750940306) in 5m0s.

Environment URL: <a href="https://deckhouse-4.test.flant.dev/products/kubernetes-platform/documentation/v1/">deckhouse-4.test.flant.dev</a>

prismagod committed:
docs: fix DVP GS review findings

- Fix blocked Next button guard (pointer-events)
- Fix step1 links and RU anchor
- Show alert when browser can't generate Ed25519 keys
- Restore recommended master resources note (#21691)
- Minor JS cleanups

Signed-off-by: Vladislav Panfilov <vladislav.panfilov@flant.com>

deckhouse-BOaTswain commented:
:green_circle: `Deploy web to test10` for `deckhouse:docs/dvp-gs-update` [succeeded](https://github.com/deckhouse/deckhouse/actions/runs/30825631302) in 12m18s.

Environment URL: <a href="https://deckhouse-10.test.flant.dev/products/kubernetes-platform/documentation/v1/">deckhouse-10.test.flant.dev</a>

deckhouse-BOaTswain commented:
:green_circle: `Deploy web to test10` for `deckhouse:docs/dvp-gs-update` [succeeded](https://github.com/deckhouse/deckhouse/actions/runs/30828443116) in 5m1s.

Environment URL: <a href="https://deckhouse-10.test.flant.dev/products/kubernetes-platform/documentation/v1/">deckhouse-10.test.flant.dev</a>

prismagod committed:
config.yaml and requirements updates

Signed-off-by: Vladislav Panfilov <vladislav.panfilov@flant.com>

deckhouse-BOaTswain commented:
:red_circle: `Deploy web to test10` for `deckhouse:docs/dvp-gs-update` [failed](https://github.com/deckhouse/deckhouse/actions/runs/31185048473) in 2m10s.

Environment URL: <a href="https://deckhouse-10.test.flant.dev/products/kubernetes-platform/documentation/v1/">deckhouse-10.test.flant.dev</a>

deckhouse-BOaTswain commented:
:red_circle: `Deploy web to test10` for `deckhouse:docs/dvp-gs-update` [failed](https://github.com/deckhouse/deckhouse/actions/runs/31185906863) in 52s.

Environment URL: <a href="https://deckhouse-10.test.flant.dev/products/kubernetes-platform/documentation/v1/">deckhouse-10.test.flant.dev</a>

deckhouse-BOaTswain commented:
:green_circle: `Deploy web to test10` for `deckhouse:docs/dvp-gs-update` [succeeded](https://github.com/deckhouse/deckhouse/actions/runs/31187311828) in 2m59s.

Environment URL: <a href="https://deckhouse-10.test.flant.dev/products/kubernetes-platform/documentation/v1/">deckhouse-10.test.flant.dev</a>

deckhouse-BOaTswain commented:
:green_circle: `Deploy web to test10` for `deckhouse:docs/dvp-gs-update` [succeeded](https://github.com/deckhouse/deckhouse/actions/runs/31374761356) in 1m0s.

Environment URL: <a href="https://deckhouse-10.test.flant.dev/products/kubernetes-platform/documentation/v1/">deckhouse-10.test.flant.dev</a>
